tools/reencode.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# The repo uses UTF-8 for text encoding, but the strings in the ROM are encoded in EUC-JP.
+# This means for matching the source must be re-encoded to EUC-JP before IDO compiles it.
+# This is conceptually equivalent to `gcc --finput-charset=UTF-8 --fexec-charset=EUC-JP`,
+# except IDO has no equivalent arguments.
+# Usage: reencode.sh [compile command minus input file...] [single input file]
+
+set -e
+
+# The last argument, the input source file to be compiled
+srcfile="${@: -1}"
+
+# Create a temporary file, and remove it on script exit
+tempfile=`mktemp --tmpdir oot_XXXXXXXX.c`
+trap "rm -f $tempfile" EXIT
+
+# Re-encode from UTF-8 to EUC-JP
+iconv -f UTF-8 -t EUC-JP -o "$tempfile" "$srcfile"
```

Comment:
sorry, this only just occurred to me when thinking about the block numbers script. Should this script add a line directive the file too for better error messages?

Reply:
Yeah I also only now realized that. working on it right now

Reply:
done but it broke matching gc-eu-mq-dbg. Reverting the bss changes in zcolchk seems to have fixed zcolchk but there's more diffs bruh. I pushed what I have rn